swift/ql/lib/codeql/swift/generated/GetImmediateParent.qll

```diff
@@ -1,240 +1,230 @@
-    or
-    exists(Expr x | optional_evaluation_exprs(this, x) and result = x)
-    or
-    exists(Pattern x | optional_some_patterns(this, x) and result = x)
-    or
-    exists(Pattern x | paren_patterns(this, x) and result = x)
-    or
-    exists(Expr x, int index | pattern_binding_decl_inits(this, index, x) and result = x)
-    or
-    exists(Pattern x, int index | pattern_binding_decl_patterns(this, index, x) and result = x)
-    or
-    exists(Expr x | rebind_self_in_constructor_exprs(this, x, _) and result = x)
-    or
-    exists(VarDecl x | rebind_self_in_constructor_exprs(this, _, x) and result = x)
-    or
-    exists(Expr x | repeat_while_stmts(this, x, _) and result = x)
-    or
-    exists(Stmt x | repeat_while_stmts(this, _, x) and result = x)
-    or
-    exists(Expr x | return_stmt_results(this, x) and result = x)
-    or
-    exists(Expr x | self_apply_exprs(this, x) and result = x)
-    or
-    exists(ConditionElement x, int index | stmt_condition_elements(this, index, x) and result = x)
-    or
-    exists(ParamDecl x, int index | subscript_decl_params(this, index, x) and result = x)
-    or
-    exists(Argument x, int index | subscript_expr_arguments(this, index, x) and result = x)
-    or
-    exists(Expr x | switch_stmts(this, x) and result = x)
-    or
-    exists(CaseStmt x, int index | switch_stmt_cases(this, index, x) and result = x)
-    or
-    exists(Expr x | tap_expr_sub_exprs(this, x) and result = x)
-    or
-    exists(BraceStmt x | tap_exprs(this, x, _) and result = x)
-    or
-    exists(Expr x | throw_stmts(this, x) and result = x)
-    or
-    exists(BraceStmt x | top_level_code_decls(this, x) and result = x)
-    or
-    exists(Expr x | tuple_element_exprs(this, x, _) and result = x)
-    or
-    exists(Expr x, int index | tuple_expr_elements(this, index, x) and result = x)
-    or
-    exists(Pattern x, int index | tuple_pattern_elements(this, index, x) and result = x)
-    or
-    exists(TypeRepr x | type_expr_type_reprs(this, x) and result = x)
-    or
-    exists(Pattern x | typed_patterns(this, x) and result = x)
-    or
-    exists(TypeRepr x | typed_pattern_type_reprs(this, x) and result = x)
-    or
-    exists(Expr x | vararg_expansion_exprs(this, x) and result = x)
-    or
-    exists(Stmt x | while_stmts(this, x) and result = x)
-    or
-    exists(Expr x, int index | yield_stmt_results(this, index, x) and result = x)
-  }
-}
+import codeql.swift.elements.Element
 
 /**
  * Gets any of the "immediate" children of `e`. "Immediate" means not taking into account node resolution: for example
  * if the AST child is the first of a series of conversions that would normally be hidden away, this will select the
  * next conversion down the hidden AST tree instead of the corresponding fully uncoverted node at the bottom.
- * This predicate is mainly intended to be used to test uniqueness of parents.
+ * Outside this module this file is mainly intended to be used to test uniqueness of parents.
```

Comment:
I think we should be more aggressive with the wording. For some reason, I can't get `Add a suggestion` to work 🤔. I'd replace this line with an initial comment saying "INTERNAL: Do not use". You'll see this [a lot](https://github.com/github/codeql/search?l=CodeQL&q=INTERNAL%3A+Do+not+use) in QL code.

Reply:
Would you place that on both the predicates, or only on `getAnImmediateChild`?

Reply:
I'd place it only on `getAnImmediateChild` as a replacement to the `Outside this module this file is mainly intended to be used to test uniqueness of parents.` part of the comment.